File: 9-palindrome-number/9-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        string palindromeString = to_string(x);
        int length = palindromeString.size() - 1;
        for (int i = 0; i < palindromeString.size(); ++i) {
           if (palindromeString.at(i) != palindromeString.at(length - i)) return false; 
        }
        return true;
    }
};

File: 9-palindrome-number/9-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0: return False
        
        left_divisor = 1
        while x >= left_divisor * 10:
            left_divisor *= 10

        while x:
            left = x  // left_divisor
            right = x % 10
            
            if left != right: return False
            
            x = (x % left_divisor) // 10
            
            left_divisor /= 100
        
        return True